src/app/backend/resource/job/detail.go

```diff
@@ -95,3 +97,27 @@ func getJobDetail(job *batch.Job, metricClient metricapi.MetricClient,
 		Completions:     job.Spec.Completions,
 	}
 }
+
+// Deletes job with given name in given namespace and related pods.
+func DeleteJob(client k8sClient.Interface, namespace, name string) error {
+
+	log.Printf("Deleting %s job from %s namespace", name, namespace)
+
+	pods, err := getRawJobPods(client, namespace, name)
+	if err != nil {
+		return err
+	}
+	if err := client.BatchV1().Jobs(namespace).Delete(name, &metaV1.DeleteOptions{}); err != nil {
```

Comment:
I think that kubectl first deletes all the pods and then deletes job with specific delete options. I'd keep their logic as they know better how it should be handled.

https://github.com/kubernetes/kubernetes/blob/a92123c5304514c2996f1c7d0f4ed4b7311cd18a/pkg/kubectl/stop.go#L351